alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -494,6 +541,31 @@ class TransportIndexMonitorAction @Inject constructor(
             if (request.monitor.enabled && currentMonitor.enabled)
                 request.monitor = request.monitor.copy(enabledTime = currentMonitor.enabledTime)
 
+            // On update monitor check which backend roles to associate to the monitor
+            if (user != null) {
+                if (request.rbacRoles != null) {
+                    if (isAdmin(user)) {
+                        request.monitor = request.monitor.copy(
+                            user = User(user.name, request.rbacRoles, user.roles, user.customAttNames)
+                        )
+                    } else {
+                        val rolesToRemove = user.backendRoles
```

Comment:
I think we could benefit from using set subtracting operations and combining calls to make this a little easier to read:

```
val rolesToRemove = user.backendRoles - request.rbacRoles.orEmpty()
...
val updatedRbac = currentMonitor.user?.backendRoles
    ?.removeAll(rolesToRemove)
    ?.addAll(rolesToAdd.orEmpty())
```

Reply:
cleaned up